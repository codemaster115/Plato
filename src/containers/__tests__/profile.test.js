import React from 'react';
import { Provider } from 'react-redux';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';

import ProfileContainer, { ProfileLayoutContainer } from '../profile';
import ProfileOverview from '../../components/Profile/ProfileOverView';

const initialState = {};
const mockStore = configureStore();
let store;

describe('ProfileLayoutContainer', () => {
  beforeEach(() => {
    store = mockStore(initialState);
  });

  it('should match snapshot', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <ProfileContainer />
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render ProfileOverView', () => {
    const wrapper = shallow(<ProfileLayoutContainer />);
    expect(wrapper.find(ProfileOverview)).toHaveLength(1);
  });
});
