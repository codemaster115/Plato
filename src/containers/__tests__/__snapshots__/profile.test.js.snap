// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileLayoutContainer should match snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeState": Object {},
    }
  }
>
  <Connect(ProfileLayoutContainer) />
</ContextProvider>
`;
